pytorch_lightning/callbacks/gpu_stats_monitor.py

```diff
@@ -101,67 +115,81 @@ def on_train_epoch_start(self, trainer, pl_module):
-            )
+            _gpu_stats['batch_time/inter_step2 (ms)'] = (time.time() - self.snap_inter_step_time) * 1000
+
+        trainer.logger.log_metrics(_gpu_stats, step=trainer.global_step)
 
         if self._log_stats.intra_step_time:
             self.snap_intra_step_time = time.time()
 
     @rank_zero_only
     def on_train_batch_end(self, trainer, pl_module, batch, batch_idx, dataloader_idx):
-        if self._log_stats.gpu_utilization:
-            self._log_usage(trainer)
-
-        if self._log_stats.memory_utilization:
-            self._log_memory(trainer)
-
-        if self._log_stats.fan_speed:
-            trainer.logger.log_metrics(self._get_gpu_stat("fan.speed", "%"), step=trainer.global_step)
-
-        if self._log_stats.temperature:
-            trainer.logger.log_metrics(self._get_gpu_stat("temperature.gpu", "degrees C"), step=trainer.global_step)
-            trainer.logger.log_metrics(self._get_gpu_stat("temperature.memory", "degrees C"), step=trainer.global_step)
+        _gpu_stat_keys = []
+        _gpu_stat_keys.extend(self._get_gpu_stat_keys())
+        _gpu_stat_keys.extend(self._get_gpu_device_stat_keys())
 
         if self._log_stats.inter_step_time:
             self.snap_inter_step_time = time.time()
 
+        _gpu_stats = self._get_gpu_stats(_gpu_stat_keys)
+
         if self._log_stats.intra_step_time and self.snap_intra_step_time:
-            trainer.logger.log_metrics(
-                {'batch_time/intra_step (ms)': (time.time() - self.snap_intra_step_time) * 1000},
-                step=trainer.global_step
-            )
+            _gpu_stats['batch_time/intra_step2 (ms)'] = (time.time() - self.snap_intra_step_time) * 1000
+
+        trainer.logger.log_metrics(_gpu_stats, step=trainer.global_step)
+
+    def _get_gpu_stats(self, gpu_stat_keys):
+        _gpu_query = ','.join([m[0] for m in gpu_stat_keys])
 
-    @staticmethod
-    def _get_gpu_stat(pitem: str, unit: str):
         result = subprocess.run(
-            [shutil.which("nvidia-smi"), f"--query-gpu={pitem}", "--format=csv,nounits,noheader"],
+            [shutil.which("nvidia-smi"), f"--query-gpu={_gpu_query}", "--format=csv,nounits,noheader"],
             encoding="utf-8",
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,  # for backward compatibility with python version 3.6
             check=True
         )
 
-        try:
-            gpu_usage = [float(x) for x in result.stdout.strip().split(os.linesep)]
-        except ValueError:
-            gpu_usage = [0]
+        def _to_float(x):
+            try:
+                return float(x)
+            except ValueError:
+                return 0.
+
+        _stats = result.stdout.strip().split(os.linesep)
+        _stats = [list(map(_to_float, x.split(', '))) for x in _stats]
+
+        _logs = {}
+        for i in range(len(_stats)):
+            _gpu_stat_keys = [f'gpu_id_{i}/gpu_{x} ({unit})' for x, unit in gpu_stat_keys]
```

Comment:
I have changed it from `gpu_stat/gpu_id` to `gpu_id/gpu_stat` so that it would display all the stats for a GPU under a single tab. Is it ok or should I revert it to just like the way it was before?

Reply:
How about tensorboard, the first is grouped per chart, right? 
so for me I would stay with GPU / metric